cmd/snap/cmd_interface.go

```diff
@@ -0,0 +1,166 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"text/tabwriter"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+
+	"github.com/jessevdk/go-flags"
+)
+
+type cmdInterface struct {
+	ShowAttrs   bool `long:"attrs"`
+	ShowAll     bool `long:"all"`
+	Positionals struct {
+		Interface interfaceName `skip-help:"true"`
+	} `positional-args:"true"`
+}
+
+var shortInterfaceHelp = i18n.G("Lists interfaces in the system")
+var longInterfaceHelp = i18n.G(`
+The interface command lists interfaces available in the system.
+
+By default a list of all used interfaces, along with a short summary, is
+displayed. Use the --all option to include unused interfaces.
+
+$ snap interfaces [--attrs] <interface>
+
+Shows details about the particular interface. The optional switch enables
+displaing of interface attributes that may be relevant to developers.
+`)
+
+func init() {
+	addCommand("interface", shortInterfaceHelp, longInterfaceHelp, func() flags.Commander {
+		return &cmdInterface{}
+	}, map[string]string{
+		"attrs": i18n.G("Show interface attributes"),
+		"all":   i18n.G("Show both used and unused interfaces"),
+	}, []argDesc{{
+		name: i18n.G("<interface>"),
+		desc: i18n.G("Show details of a specific interface"),
+	}})
+}
+
+func (x *cmdInterface) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	if x.Positionals.Interface != "" {
+		// Show one interface in detail.
+		name := string(x.Positionals.Interface)
+		iface, err := Client().Interface(name)
+		if err != nil {
+			return err
+		}
+		x.showOneInterface(&iface)
+	} else {
+		// Show an overview of available interfaces.
+		ifaces, err := Client().InterfaceIndex()
+		if err != nil {
+			return err
+		}
+		if len(ifaces) == 0 {
+			return fmt.Errorf(i18n.G("no interfaces found"))
+		}
+		x.showManyInterfaces(ifaces)
+	}
+	return nil
+}
+
+func (x *cmdInterface) showOneInterface(iface *client.Interface) {
+	w := tabwriter.NewWriter(Stdout, 2, 2, 1, ' ', 0)
+	defer w.Flush()
+
+	fmt.Fprintf(w, "name:\t%s\n", iface.Name)
+	if iface.Summary != "" {
+		fmt.Fprintf(w, "summary:\t%s\n", iface.Summary)
+	}
+	if iface.Description != "" {
+		// FIXME: find out for real
+		termWidth := 77
+		fmt.Fprintf(w, "description: |\n%s\n", formatDescr(iface.Description, termWidth))
+	}
+	if iface.DocsURL != "" {
+		fmt.Fprintf(w, "docs-url:\t%s\n", iface.DocsURL)
```

Comment:
I think this can say:
```
("documentation:\t%s\n", iface.DocURL)
```
Perhaps even dropping those \t?  Not sure if they're helping much in this case.